pkg/openSearch/openSearchQuery/compareHandler.go

```diff
@@ -78,37 +78,31 @@ func handleCompareOperatorContainsDifferent(fieldName string, fieldKeys []string
 
 // HandleCompareOperatorBeginsWith handles begins with
 func HandleCompareOperatorBeginsWith(fieldName string, fieldKeys []string, fieldValue any, querySettings *QuerySettings) esquery.Mappable {
-	// for list of values
-	if querySettings.WildcardArrays != nil &&
-		querySettings.WildcardArrays[fieldName] {
-		return handleCompareOperatorBeginsWithDifferent(fieldName, fieldKeys, fieldValue, querySettings)
-	} else {
-		if values, ok := fieldValue.([]interface{}); ok {
-			return esquery.Bool().
-				Should(
-					lo.Map[interface{}, esquery.Mappable](values, func(value interface{}, _ int) esquery.Mappable {
-						return esquery.Prefix(fieldName+".keyword", valueToString(value))
-					})...,
-				).
-				MinimumShouldMatch(1)
-		} else { // for single values
-			return esquery.Prefix(fieldName+".keyword", valueToString(fieldValue))
-		}
+	isWildcardArray := querySettings.WildcardArrays != nil && querySettings.WildcardArrays[fieldName]
+	field := fieldName + ".keyword"
+
+	// if the query settings specify that it is a wildcard array, use the field name appended with the '.keyword'
```

Comment:
If I read the code correctly, it is the other way round. For wildcard arrays the no `.keyword` is appended

Reply:
True. 
The comments in the code has been adjusted